Ui home page implementation

Kevin-Guillen opened the pull request:
Ui home page implementation where i get two endpoints. "Trending" for the swiper and "Discover" for the grid.

https://user-images.githubusercontent.com/84336422/123908562-10f37800-d94e-11eb-8a76-17f7cc184a9a.mp4

Kevin-Guillen committed:
Implemented the ui of the homepage.

marianosalvetti commented on lib/src/bloc/movies_bloc.dart:
```diff
@@ -7,23 +7,34 @@ import 'i_movies_bloc.dart';
 class MoviesBloc extends IMoviesBloc {
   MoviesRepository _moviesRepository = MoviesRepository();
   StreamController<Movies> _moviesStreamController = StreamController();
+  StreamController<Movies> _swiper = StreamController();
```
swiper? ponele un nombre màs descriptivo ;) que hace esa variable?

Kevin-Guillen replied:
done

marianosalvetti commented on lib/src/bloc/movies_bloc.dart:
```diff
@@ -7,23 +7,34 @@ import 'i_movies_bloc.dart';
 
   @override
-  Stream<Movies> get stream => this._moviesStreamController.stream;
+  Stream<Movies> get stream => _moviesStreamController.stream;
```
moviesStream maybe? nombre de esa variable.

Kevin-Guillen replied:
done

marianosalvetti commented on lib/src/resources/movies_api_provider.dart:
```diff
@@ -1,28 +1,47 @@
+    final trendingResponse = await client.get(
       Uri.parse(
-        Constants.uriClient + Constants.apiKey,
+        MovieStrings.trendingMoviesEndpoint + MovieStrings.apiKey,
```
bien !

marianosalvetti commented on lib/src/utils/ui_constants.dart:
```diff
@@ -0,0 +1,22 @@
+abstract class UiConstants {
```
bien :)

FranciscoGaraventa commented on lib/src/model/movie_result.dart:
```diff
@@ -1,3 +1,5 @@
+import 'package:movies_project/src/utils/movie_string.dart';
```
You missed a relative import here

Kevin-Guillen replied:
done

FranciscoGaraventa commented on lib/src/resources/movies_api_provider.dart:
```diff
@@ -1,28 +1,47 @@
 import 'dart:async';
 import 'dart:convert';
 import 'package:http/http.dart' as http;
+import 'package:movies_project/src/utils/movie_string.dart';
```
You missed a relative import here

Kevin-Guillen replied:
done

FranciscoGaraventa commented on lib/src/ui/movies_page.dart:
```diff
@@ -1,5 +1,9 @@
 import 'package:flutter/material.dart';
+import 'package:movies_project/src/bloc/movies_bloc.dart';
+import 'package:movies_project/src/model/movies.dart';
```
You missed relative imports here

Kevin-Guillen replied:
done

FranciscoGaraventa commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                ),
+                child: Padding(
+                  padding:
+                      const EdgeInsets.only(top: UiConstants.appBarTopPadding),
```
You missed a trailing comma here

Kevin-Guillen replied:
added

FranciscoGaraventa commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                          right: UiConstants.searchTextPadding,
+                          top: UiConstants.searchTextPadding),
+                      filled: true,
+                      fillColor: Colors.white12),
```
You missed a trailing comma here

Kevin-Guillen replied:
added

FranciscoGaraventa commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+      ),
+      body: SingleChildScrollView(
+        child: Padding(
+          padding: const EdgeInsets.all(UiConstants.bodyPadding),
```
You missed a trailing comma here

Kevin-Guillen replied:
added

FranciscoGaraventa commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                alignment: Alignment.centerLeft,
+                child: Padding(
+                  padding: const EdgeInsets.only(
+                      bottom: UiConstants.movieTypeTextPadding),
```
You missed a trailing comma here

Kevin-Guillen replied:
added

FranciscoGaraventa commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                          child: Padding(
+                            padding: const EdgeInsets.only(
+                                top: UiConstants
+                                    .swiperProgressIndicatorTopPadding),
```
You missed a trailing comma here

Kevin-Guillen replied:
added

FranciscoGaraventa commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                        )
+                      : Padding(
+                          padding: const EdgeInsets.only(
+                              top: UiConstants.gridProgressIndicatorTopPadding),
```
You missed a trailing comma here

Kevin-Guillen replied:
added

FranciscoGaraventa commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+    );
+  }
+
+  Widget _swiper(AsyncSnapshot snapshot) {
+    return Container(
+        height: UiConstants.swiperCardsHeight,
+        child: Container(
+          child: Swiper(
+            itemBuilder: (BuildContext context, int index) {
+              return new Image.network(
+                snapshot.data.results[index].posterPath,
+                fit: BoxFit.fill,
+              );
+            },
+            itemCount: snapshot.data.results.length,
+            itemWidth: UiConstants.swiperCardsWidth,
+            layout: SwiperLayout.STACK,
+          ),
+        ));
+  }
```
You can abstract the swiper widget into a separate file.

Kevin-Guillen replied:
done

MarcosCGlobant commented on lib/src/bloc/i_movies_bloc.dart:
```diff
@@ -6,5 +6,9 @@ import 'bloc.dart';
 abstract class IMoviesBloc extends Bloc {
   Stream<Movies> get stream;
 
-  void fetchMovies();
+  Stream<Movies> get swiper;
```
try to find more descriptive names for this streams accordingly with what they do, and apply that on the movie bloc class, to keep consistency

Kevin-Guillen replied:
done

MarcosCGlobant commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                    style: TextStyle(
+                        fontSize: UiConstants.movieTypeTitleFontSize,
+                        color: Colors.grey,
+                        fontFamily: 'Play-Bold'),
```
missed a trailing comma here

Kevin-Guillen replied:
added

MarcosCGlobant commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                              strokeWidth:
+                                  UiConstants.progressIndicatorStrokeWidth,
+                              backgroundColor: Colors.purple,
+                              valueColor: new AlwaysStoppedAnimation<Color>(
+                                  Colors.black),
```
the "new" identifier is not necessary, and missed a trailing comma on the second line

Kevin-Guillen replied:
removed

MarcosCGlobant commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                              strokeWidth:
+                                  UiConstants.progressIndicatorStrokeWidth,
+                              backgroundColor: Colors.purple,
+                              valueColor: new AlwaysStoppedAnimation<Color>(
+                                  Colors.black),
```
the "new" identifier is not necessary, and missed a trailing comma on the second line

Kevin-Guillen replied:
removed and added

MarcosCGlobant commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+        child: Container(
+          child: Swiper(
+            itemBuilder: (BuildContext context, int index) {
+              return new Image.network(
```
the "new" identifier is not necessary

Kevin-Guillen replied:
removed

MarcosCGlobant commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                child: Padding(
+                  padding: const EdgeInsets.only(
+                      top: UiConstants.movieTypeTextPadding,
+                      bottom: UiConstants.movieTypeTextPadding),
```
missed a trailing comma

Kevin-Guillen replied:
added

MarcosCGlobant commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                  padding: const EdgeInsets.only(
+                      top: UiConstants.movieTypeTextPadding,
+                      bottom: UiConstants.movieTypeTextPadding),
+                  child: new Text(
```
the "new" identifier is not necessary

Kevin-Guillen replied:
removed

MarcosCGlobant commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                    style: TextStyle(
+                        fontSize: UiConstants.movieTypeTitleFontSize,
+                        color: Colors.grey,
+                        fontFamily: 'Play-Bold'),
```
missed a trailing comma

Kevin-Guillen replied:
added

MarcosCGlobant commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                  shrinkWrap: true,
+                  physics: ScrollPhysics(),
+                  itemCount: snapshot.data.results.length,
+                  gridDelegate: new SliverGridDelegateWithFixedCrossAxisCount(
```
the "new" identifier is not necessary

Kevin-Guillen replied:
removed

MarcosCGlobant commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                        child: Container(
+                          child: ClipRRect(
+                            borderRadius: BorderRadius.only(
+                              topLeft: Radius.circular(
+                                  UiConstants.gridCardsBorderRadius),
+                              bottomRight: Radius.circular(
+                                  UiConstants.gridCardsBorderRadius),
```
missed trailing commas at the end of both of the radiuses

Kevin-Guillen replied:
added

MarcosCGlobant commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+              Padding(
+                padding: const EdgeInsets.only(
+                    left: UiConstants.searchBarSidesPadding,
+                    right: UiConstants.searchBarSidesPadding),
```
missed a trailing comma here

Kevin-Guillen replied:
added

MarcosCGlobant commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                    style: TextStyle(
+                        color: Colors.purple,
+                        fontSize: UiConstants.titleFontSize,
+                        fontFamily: 'Play-Bold'),
```
missed a trailing comma here

Kevin-Guillen replied:
added

MarcosCGlobant commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                  style: TextStyle(
+                      color: Colors.grey,
+                      fontSize: UiConstants.searchTextFontSize,
+                      fontFamily: 'Play-Bold'),
```
missed a trailing comma here

Kevin-Guillen replied:
added

MarcosCGlobant commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                      contentPadding: const EdgeInsets.only(
+                          left: UiConstants.searchTextPadding,
+                          right: UiConstants.searchTextPadding,
+                          top: UiConstants.searchTextPadding),
```
missed a trailing comma here

Kevin-Guillen replied:
added

MarcosCGlobant commented on lib/src/ui/movies_page.dart:
```diff
@@ -14,11 +18,225 @@ class _MoviesPageState extends State<MoviesPage> {
+                child: Padding(
+                  padding: const EdgeInsets.only(
+                      bottom: UiConstants.movieTypeTextPadding),
+                  child: new Text(
```
the "new" identifier is not necessary

Kevin-Guillen replied:
removed

MarcosCGlobant commented on lib/src/widgets/gridview.dart:
```diff
@@ -0,0 +1,78 @@
+            : CircularProgressIndicator(
+                strokeWidth: UiConstants.progressIndicatorStrokeWidth,
+                backgroundColor: Colors.purple,
+                valueColor: new AlwaysStoppedAnimation<Color>(Colors.black),
```
the "new" identifier is not necessary, and missed a trailing comma on the second line

Kevin-Guillen replied:
removed

MarcosCGlobant commented on lib/src/widgets/card_swiper.dart:
```diff
@@ -0,0 +1,58 @@
+              : CircularProgressIndicator(
+                  strokeWidth: UiConstants.progressIndicatorStrokeWidth,
+                  backgroundColor: Colors.purple,
+                  valueColor: new AlwaysStoppedAnimation<Color>(Colors.black),
```
the "new" identifier is not necessary, and missed a trailing comma on the second line

Kevin-Guillen replied:
removed

Kevin-Guillen committed:
fixing comments :

- added missing trailing comma
- widgets abstraction
- deleted unnecessary "new" identifier
- changed non-descriptive names

MarcosCGlobant reviewed (approved):
LGTM

FranciscoGaraventa commented on lib/src/model/movie_result.dart:
```diff
@@ -43,10 +45,11 @@ class MoviesResult {
       title: parsedJson['title'],
       popularity: parsedJson['popularity'],
-      posterPath: parsedJson['poster_path'],
+      posterPath: parsedJson['poster_path'] != null
+          ? MovieStrings.imagesPath + parsedJson['poster_path']
+          : MovieStrings.defaultImage,
```
Take a look at this null operator: https://news.dartlang.org/2015/08/dart-112-released-with-null-aware.html, this can make you save coding time a it will make your code easy to read

Kevin-Guillen replied:
I took a look at it. It was useful for me to learn these types of operators but in this case I cannot use any of those because  the expression will never resolve null.

FranciscoGaraventa reviewed (approved):
LGTM
